velox/type/DecimalUtil.h

```diff
@@ -32,5 +32,34 @@ class DecimalUtil {
   /// Helper function to convert a decimal value to string.
   template <typename T>
   static std::string toString(const T& value, const TypePtr& type);
+
+  inline static int128_t rescaleWithRoundUp(
+      const int128_t& unscaledValue,
+      const uint8_t fromScale,
+      const std::pair<int, int>& toPrecisionScale) {
+    auto rescaledValue = unscaledValue;
+    auto scaleDifference = toPrecisionScale.second - fromScale;
+    if (scaleDifference >= 0) {
+      rescaledValue *= DecimalUtil::kPowersOfTen[scaleDifference];
+    } else {
+      scaleDifference = -scaleDifference;
+      rescaledValue =
+          unscaledValue / DecimalUtil::kPowersOfTen[scaleDifference];
+      int128_t remainder =
+          unscaledValue % DecimalUtil::kPowersOfTen[scaleDifference];
+      if (unscaledValue >= 0 &&
```

Comment:
can we use the `std::round()` function here?

Reply:
https://en.cppreference.com/w/cpp/numeric/math/round.
I tried with lround(), didn't get right values, returned 0. No support for int128_t.

Reply:
you can make the remainder a `double` type?

Reply:
This is fine. `Double` will lose some precision.